pkg/controller/perconaservermongodb/getters.go

```diff
@@ -117,8 +118,18 @@ func (r *ReconcilePerconaServerMongoDB) getMongodStatefulsets(cr *api.PerconaSer
 	return list, err
 }
 
+func (r *ReconcilePerconaServerMongoDB) matchUID(cr *api.PerconaServerMongoDB, obj metav1.Object) bool {
+	if ref := metav1.GetControllerOf(obj); ref != nil {
+		if string(cr.GetUID()) == string(ref.UID) {
```

Comment:
You don't need to compare UID's manually
All can be done via `metav1.IsControlledBy(obj,cr)`